vendor/speedscope/update.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```

Comment:
Could this possibly turned into a Ruby script/`rake` task?

The `Gem::Package::TarReader` should be able to handle the `Tar` portion of things (cross platform), and then you can basically use `FileUtils` for the rest.

Obviously `npm` is still required and would need a shell out, but that is just a given and as you have stated it is just a developer dependency.

Reply:
It could be, but I'm not sure it's worth doing -- is the target benefit to make it possible for a Windows-only maintainer to run the update script?

Reply:
So there are a few non-"but what about WIN-DOSE" reasons I tend to always push for this myself, and it very much is pedantic in most cases... BUT YOU ASKED FOR IT! (he says sarcastically):

(I am writing a lot here, but it's again, this is just an explanation to my personal opinionated stance and just what I do myself and why.  Zero pressure to make the change and feel free to take it or leave it.) 

* Shelling out has some cost associated with it.  Next to nothing 99% of the time, but depending on how much you need to do it, it can add up.
* When I use "Windows" as an example, bit is more about removing inconsistencies between platforms, and since we know Ruby is going to be used by who ever is running this script, we can assume it as a more consistent constant.  For a few examples in your case:
  * A user might have some weird version of `/bin/bash` (probably a bad example...)
  * `which` might not exist on the `$PATH` on some machines, or the user might be running with `sudo`
  * weird user aliases being loaded
* Specifically to the `rake` suggestion, this is simply for dev/maintainer consistency.  I know that that did put a README together, but from personal experience, most people don't RTFM.  But assuming they would at least do a `rake -T` when they clone the project, they can get at a glance what general dev tasks are available.
  - Further more, if it is written in ruby, we then are just doing `Ruby -> npm` for that one time we need to shell out, instead of `Ruby -> shell -> npm/tar/...`.  Less moving parts and less places for failure.

`</2cents>`

Reply:
@tenderlove Do you have opinions on this front? I don't spend too much time in the ruby ecosystem, so I'm not sure what the expectations are on this front. I'm happy to do this if you feel strongly, but would otherwise bias towards not doing it, since not doing it less work 😅

Reply:
@jlfwong I think you meant to direct this at me, but I can put something together for you if it is decided this would be preferred.

Reply:
Sorry, I misread the past comment and just noticed your true intent.  Sorry.

I did put together this to show how this could be done, but again, no pressure to actually implement this, and it was mostly a interesting exercise for myself.

```ruby
# Rakefile  (I personally put this starting around line 11)

def untar(tarfile)
  # Little put of a hack with the `.new` to get this to work
  Gem::Package.new("").open_tar_gz tarfile do |tar|
    tar.each { |file| yield file }
  end
end

desc "Update speedscope assets"
task :update_speedscope do
  rm_rf   "vendor/speedscope"
  mkdir_p "vendor/speedscope"

  cd "vendor/speedscope" do
    sh "npm pack speedscope"

    File.open Dir.glob("speedscope*.tgz").first do |tarfile|
      untar tarfile do |file|
        next unless file.full_name =~ /^package\/(LICENSE|dist\/release\/.*(html|css|js|png))$/
        puts file.full_name

        File.write File.basename(file.full_name), file.read
      end

      rm tarfile.path
    end
  end
```

One note, this does change the scoping of the resulting directory structure from `vendor/speedscope/speedscope` to just `vendor/speedscope`, since the nested dir seemed redundant (but I could be missing something).  I would assume that some changes in `report.rb` would be necessary if this is a reasonable change, otherwise a few tweaks to this task could be made to emulate what already exists.

Reply:
Thanks for writing up sample code! I always appreciate it when people are willing to take the time to write code to support their ideas.

That said, I'm not planning on changing this to a rake task unless this is considered a blocker for merging.

My reasoning here is that I'm intending to make similar changes to several different profilers in many different languages (e.g. pyflame), and would prefer the update scripts to look as similar as possible. Changing them to be specific to the language of the profiler would make that more difficult.